tasks/unit_tests/components_tests.py

```diff
@@ -29,41 +38,135 @@ def test_find_team(self):
         self.assertEqual(teamname, 'agent-shared-components')
 
     def test_get_components_and_bundles(self):
-        results = components.get_components_and_bundles()
-        bundles = results[0]
-        comps = results[1]
-        ok = results[2]
-        self.assertEqual([], bundles)
-        self.assertEqual(3, len(comps))
+        comps, bundles = components.get_components_and_bundles()
+        self.assertEqual(1, len(bundles))
+        self.assertEqual('comp/group', bundles[0].path)
+        self.assertEqual(1, len(bundles[0].components))
+        self.assertEqual('comp/group/inbundle', bundles[0].components[0].path)
+        self.assertEqual(4, len(comps))
         self.assertEqual('comp/classic', comps[0].path)
-        self.assertEqual('comp/multiple', comps[1].path)
-        self.assertEqual('comp/newstyle', comps[2].path)
-        self.assertEqual(ok, False)
+        self.assertEqual('comp/group/inbundle', comps[1].path)
+        self.assertEqual('comp/multiple', comps[2].path)
+        self.assertEqual('comp/newstyle', comps[3].path)
 
-        # Add the classic component (used by tests) to the allowlist
-        # To validate that calling get_components_and_bundles does not return an error
-        classicComp = 'comp/classic/classicimpl'
-        if classicComp not in components.components_classic_style:
-            components.components_classic_style.append(classicComp)
+    def test_validate_bundles(self):
+        _, bundles = components.get_components_and_bundles()
+        errs = components.validate_bundles(bundles)
+        self.assertEqual(0, len(errs))
+
+        # Lint error because bundle defines an interface
+        filename = os.path.join(bundles[0].path, 'bundle.go')
+        append_line(filename, 'type Component interface{}')
+
+        _, bundles = components.get_components_and_bundles()
+        errs = components.validate_bundles(bundles)
+        self.assertEqual(1, len(errs))
+        self.assertIn('Component interface', errs[0])
+
+        # No lint errors with original source
+        self.copy_component_src_to_tmpdir()
```

Comment:
What about renaming to something that signal that we are resetting the initial component folders? `self.set_component_test_data_to_tmpdir()` (Maybe)

Reply:
Decided to go with `reset_component_src_in_tmpdir`.